sdk/main/src/AccountUpdateTransaction.cc

```diff
@@ -138,6 +155,16 @@ AccountUpdateTransaction& AccountUpdateTransaction::setStakedNodeId(const uint64
   return *this;
 }
 
+//-----
+AccountUpdateTransaction& AccountUpdateTransaction::clearStakedNodeId()
+{
+  requireNotFrozen();
+
+  mStakedNodeId = -1LL;
```

Comment:
Same as `mStakedAccountId` - since `mStakedNodeId` is optional, shouldn't we set it to `std::nullopt` when clearing the `StakedNodeId`?

Reply:
Same explanation applies here as it does for [`mAccountMemo`](https://github.com/hashgraph/hedera-sdk-cpp/pull/558#discussion_r1368755086).